src/operations.cc

```diff
@@ -351,6 +351,31 @@ namespace sharp {
     }
   }
 
+  /*
+   * Unflatten
+   */
+  VImage Unflatten(VImage image, std::vector<double> const thresholds) {
+    if (HasAlpha(image)) {
```

Comment:
Would we ever want to discard the existing alpha channel?

Reply:
Likely not :) The new implementation will convert white to transparent, and leave non-white alpha values as is.